.github/workflows/deps-test.yml

```diff
@@ -95,3 +63,181 @@ jobs:
           title: Dependency Test Report
           content-filepath: ./deps-test.md
           labels: report, automated issue
+
+  compile:
+    # Use 20.04.5 until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16450 is resolved
+    runs-on: ubuntu-20.04
+    needs: [ update-deps ]
+    strategy:
+      matrix:
+        SYS_BINARIES: [ "binaries-darwin_amd64", "binaries-linux_amd64", "binaries-linux_arm64", "binaries-windows_amd64", "binaries-linux_ppc64le" ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Restore cache from update-deps
+        id: restore-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            **/go.mod
+            **/go.sum
+            ~/go/pkg/mod
+          key: update-deps-${{ github.run_id }}
+      - uses: actions/setup-go@v3
+        if: steps.restore-cache.outputs.cache-hit == 'true'
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: make ${{ matrix.SYS_BINARIES }}
+        if: steps.restore-cache.outputs.cache-hit == 'true'
+      - uses: actions/upload-artifact@v3
+        if: steps.restore-cache.outputs.cache-hit == 'true'
+        with:
+          name: ${{ matrix.SYS_BINARIES }}
+          path: bin/
+
+  unit-tests:
+    # Use 20.04.5 until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16450 is resolved
+    runs-on: ubuntu-20.04
+    needs: [ update-deps ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Restore cache from update-deps
+        id: restore-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            **/go.mod
+            **/go.sum
+            ~/go/pkg/mod
+          key: update-deps-${{ github.run_id }}
+      - uses: actions/setup-go@v3
+        if: steps.restore-cache.outputs.cache-hit == 'true'
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: make test
+        if: steps.restore-cache.outputs.cache-hit == 'true'
```

Comment:
Wait `cache-hit=='true'` means we should build things?  Wouldn't it be the opposite (`cache-hit != 'true'` )?

Reply:
For this workflow, we only need to run the tests when there are new upstream core/contrib releases.  After running the `update-deps` script, it checks if the go.mod/go.sum files are changed.  These files are then cached only if there are changes, and the downstream build/test jobs only run when there is a cache hit.